app/scripts/modules/core/src/presentation/HoverablePopover.tsx

```diff
@@ -17,21 +23,75 @@ export interface IProps {
 export interface IState {
   popoverIsOpen: boolean;
   target?: any;
+  PopoverRenderer?: React.ComponentClass<PopoverProps> | React.StatelessComponent<PopoverProps>;
 }
 
 @autoBindMethods
 export class HoverablePopover extends React.Component<IProps, IState> {
   public static defaultProps: Partial<IProps> = {
     placement: 'top',
-    id: UUIDGenerator.generateUuid()
+    id: UUIDGenerator.generateUuid(),
   };
 
   private popoverCancel: number;
 
+  /**
+   * Renders a Popover offset to the left or right.
+   * offetPercent: a percent string between 0 and 99, e.g.: `50%`, `12%` or `90%`
+   */
+  private static popoverOffsetRenderer(offsetPercent: string) {
+    interface IOffsetState {
+      offset: number,
+    }
+
+    return class extends React.Component<PopoverProps, IOffsetState> {
+      public state: IOffsetState = {
+        offset: null,
```

Comment:
typescript complains if you omit the `offset: null`.  Either of these would work:

```
      public state: IOffsetState = {
        offset: null,
      };
```

```
      public state = {} as IOffsetState;
```